src/js/components/Drop/StyledDrop.js

```diff
@@ -27,6 +27,88 @@ const dropKeyFrames = keyframes`
 
+const isObject = object => object != null && typeof object === 'object';
+
+const deepEqual = (object1, object2) => {
+  const keys1 = Object.keys(object1);
+  const keys2 = Object.keys(object2);
+
+  if (keys1.length !== keys2.length) {
+    return false;
+  }
+
+  for (let i = 0; i < keys1.length; i += 1) {
+    const val1 = object1[keys1[i]];
+    const val2 = object2[keys1[i]];
+    const areObjects = isObject(val1) && isObject(val2);
+    if (
+      (areObjects && !deepEqual(val1, val2)) ||
+      (!areObjects && val1 !== val2)
+    ) {
+      return false;
+    }
+  }
+  return true;
+};
+
+const marginStyle = (theme, align, data, responsive, margin) => {
+  const marginProp = theme.global.edgeSize[data] || data;
+  let MarginSide;
+
+  if (theme.global.drop.intelligentMargin === true) {
+    if (deepEqual(align, { top: 'bottom' })) {
+      MarginSide = { top: marginProp };
+    } else if (deepEqual(align, { bottom: 'top' })) {
+      MarginSide = { bottom: marginProp };
+    } else if (deepEqual(align, { right: 'left' })) {
+      MarginSide = { left: `-${marginProp}` };
+    } else if (deepEqual(align, { left: 'right' })) {
+      MarginSide = { left: marginProp };
+    } else if (deepEqual(align, { top: 'top', left: 'right' })) {
+      MarginSide = { left: marginProp };
+    } else if (deepEqual(align, { top: 'top', right: 'left' })) {
+      MarginSide = { left: `-${marginProp}` };
+    } else if (deepEqual(align, { top: 'bottom', right: 'left' })) {
+      MarginSide = { top: marginProp, left: `-${marginProp}` };
+    } else if (deepEqual(align, { top: 'bottom', right: 'right' })) {
+      MarginSide = { top: marginProp };
+    } else if (deepEqual(align, { top: 'bottom', left: 'right' })) {
+      MarginSide = { top: marginProp, left: marginProp };
+    } else if (deepEqual(align, { top: 'bottom', left: 'left' })) {
+      MarginSide = { top: marginProp };
+    } else if (deepEqual(align, { bottom: 'top', right: 'left' })) {
+      MarginSide = { bottom: marginProp, left: `-${marginProp}` };
+    } else if (deepEqual(align, { right: 'right', bottom: 'top' })) {
+      MarginSide = { bottom: marginProp };
+    } else if (deepEqual(align, { bottom: 'top', left: 'left' })) {
+      MarginSide = { bottom: marginProp };
+    } else if (deepEqual(align, { bottom: 'top', left: 'right' })) {
+      MarginSide = { bottom: marginProp, left: marginProp };
+    } else if (deepEqual(align, { bottom: 'top', right: 'left' })) {
+      MarginSide = { bottom: marginProp, left: `-${marginProp}` };
+    } else if (deepEqual(align, { bottom: 'bottom', left: 'right' })) {
+      MarginSide = { left: marginProp };
+    } else if (deepEqual(align, { bottom: 'bottom', right: 'left' })) {
+      MarginSide = { left: `-${marginProp}` };
+    }
+  } else {
+    return edgeStyle(
+      'margin',
+      margin || theme.global.drop.margin,
+      responsive,
+      theme.global.edgeSize.responsiveBreakpoint,
+      theme,
+    );
+  }
+  return edgeStyle(
```

Comment:
When would this return statement be reached? it seems like the return at 95 would always be the last

Reply:
The return at line 95 is for when intelligentMargin is not set to true..